hack/dockerized

```diff
@@ -24,12 +31,19 @@ fi
 # Make sure that the output directory exists
 docker run -v "${BUILDER}:/root:rw,z" --rm ${BUILDER} mkdir -p /root/go/src/kubevirt.io/kubevirt/_out
 
+# Add all of the environment variables that may be needed within the build container
+append_envvar_file KUBEVIRT_PROVIDER ${KUBEVIRT_PROVIDER}
```

Comment:
In order to inject these values for the external provider, it should be ablot to create your own config file, like we do e.g. here: https://github.com/kubevirt/kubevirt/blob/master/cluster/local/provider.sh#L35. The name of the file should be `hack/config-provider-external.sh`.

You should not have to modify `hack/dockerized`.

Reply:
This doesn't work for me.  When a script is run "dockerized", for example "build_manifest.sh", unless you pass the provider in as an environment variable, the script doesn't know its provider and thus the per-provider script will not get called.
```
Before calling, in dockerized KUBEVIRT_PROVIDER=external
go version go1.10 linux/amd64
+ source hack/common.sh
++++ dirname 'hack/common.sh[0]'
+++ cd hack/../
+++ pwd
++ KUBEVIRT_DIR=/root/go/src/kubevirt.io/kubevirt
++ OUT_DIR=/root/go/src/kubevirt.io/kubevirt/_out
++ VENDOR_DIR=/root/go/src/kubevirt.io/kubevirt/vendor
++ CMD_OUT_DIR=/root/go/src/kubevirt.io/kubevirt/_out/cmd
++ TESTS_OUT_DIR=/root/go/src/kubevirt.io/kubevirt/_out/tests
++ APIDOCS_OUT_DIR=/root/go/src/kubevirt.io/kubevirt/_out/apidocs
++ MANIFESTS_OUT_DIR=/root/go/src/kubevirt.io/kubevirt/_out/manifests
++ MANIFEST_TEMPLATES_OUT_DIR=/root/go/src/kubevirt.io/kubevirt/_out/templates/manifests
++ PYTHON_CLIENT_OUT_DIR=/root/go/src/kubevirt.io/kubevirt/_out/client-python
++ KUBEVIRT_PROVIDER=k8s-1.10.3
++ KUBEVIRT_NUM_NODES=1
```

Part of the problem is that the same "dockerized" environment is being used for provider-independent things like building binaries, and for provider-dependent things like building the manifests.  IMHO nothing run in this particular container should have to know about the test environment.  Maybe only build templatized (i.e. Jinja2) manifests inside the container, then apply the data outside, i.e. in the provider-aware environment.  I'll try to get something like this working.

Reply:
> This doesn't work for me. When a script is run "dockerized", for example "build_manifest.sh", unless you pass the provider in as an environment variable, the script doesn't know its provider and thus the per-provider script will not get called.

Ah yes you are right. What we do is something like this for other providers in the `build` function of the provider:

```
${KUBEVIRT_PATH}hack/dockerized KUBEVIRT_PROVIDER=${KUBEVIRT_PROVIDER} ./hack/build-manifests.sh"
```

> IMHO nothing run in this particular container should have to know about the test environment. Maybe only build templatized (i.e. Jinja2) manifests inside the container, then apply the data outside, i.e. in the provider-aware environment. I'll try to get something like this working.

I see the difficulty with passing in the data. However it is very valuable that everything runs in containers. Maybe the suggestion above is good enough. Although I see that it is not perfect. Maybe we can wait with refactoring until we see the need for more providers? I hope that we don't need much more. The `external` provider seems to be the generic missing one.